### DISCUSSION TITLE
Responses to bot commands

### DISCUSSION THREAD

**Discussion by MrPRAINGLE** [discordjs/discord.js#6955] [Q&A] [answered]
Hey !

Recently, when I was coding, I went in a trouble. In the bot that I actually use (so not the mine), it is possible to answer commands. For an example, you can type "!clear" and the bot asks you how many messages you want to delete. If you answer by a number, it will delete as any messages as specified by this number.

I tried with the "Client.on("message")" technique, but the problem is that you can only response once. Could you help me please ?

**Comment by ImRodry** [comment, marked as answer]
You should try using [awaitMessages](https://discord.js.org/#/docs/main/stable/class/TextChannel?scrollTo=awaitMessages)

**Reply by MrPRAINGLE** [reply]
Ye, I saw that before, but I don't know what to put in filter... Do I need to put the responses I want ? (I can't try right now, that's why I' asking)

Edit : I just foud a tutorial on [stackoverflow](https://stackoverflow.com/questions/68981071/how-to-use-awaitmessages-in-discord-js-v13), in filter I need to put "message_filter, max: 1" and then I see the responses.

Edit 2 : But there is, in fact, still a problem : if the answer is not a thing that we want, the code stops anyways... Is there a way to come back, like a return but at a precise moment in the script ?

**Reply by ImRodry** [reply]
The filter is a function that filters unwanted messages and gives you only the ones you want. You often want to filter by id or something, depends on your use case

**Reply by ghost** [reply]
> Ye, I saw that before, but I don't know what to put in filter... Do I need to put the responses I want ? (I can't try right now, that's why I' asking)
> 
> Edit : I just foud a tutorial on [stackoverflow](https://stackoverflow.com/questions/68981071/how-to-use-awaitmessages-in-discord-js-v13), in filter I need to put "message_filter, max: 1" and then I see the responses.
> 
> Edit 2 : But there is, in fact, still a problem : if the answer is not a thing that we want, the code stops anyways... Is there a way to come back, like a return but at a precise moment in the script ?

It will not return back unless you run the command again.

**Reply by MrPRAINGLE** [reply]
Thanks ! But I meant to like... tp send a message such as "This is not an ID" if the wanted value is a snowflake

**Reply by ghost** [reply]
```js
const msg = await client.channels.cache.get(channelid).fetchMessage(messageid);

if(!msg) {
    return msg.channel.send("Invalid Snowflake Id")
}
```

Could be resolved like this. This is not the best code but you can improve it.

Note: Its DJS V12

**Reply by ghost** [reply]
```js
const msg = await msg.channel.messages.fetch(messageid);

if(!msg){
    return msg.channel.send("Invalid Snowflake ID");
} 

const deletedMessage = await msg.delete();

```
This one is the solution for the **Current Stable Version**. This is not a definite solution because improvements could be made.

**Reply by MrPRAINGLE** [reply]
It's a good method, but if we do a ``!ban`` + <another thing that a snowflake>, the bot answers ``This not a snowflake, idiot. try again" and I want that if the user sends a correct snowflake (without ``!ban``) the user that corresponds to the snowflake get banned... I saw this on several bots but I don't know how does that works

**Reply by ghost** [reply]
You could change the maximum amount messages needed to be collected. Use the `max` property for setting the maximum messages. You could remove the property for infinite message collection. 

eg:
```js
msg.channel.awaitMessages({ filter, max: 4,})
```

**Comment by ghost** [comment]
Hi, I was wondering if your issue has been resolved?

@MrPRAINGLE
